File: skill_sdk/tests/fixtures/js-project/src/shop/payment/MockGateway.js
import { PaymentGateway } from './PaymentGateway.js';
import { PaymentResult } from './PaymentResult.js';

/**
 * Always-succeeds mock gateway for testing.
 *
 * @implements {PaymentGateway}
 */
export class MockGateway extends PaymentGateway {
  constructor() {
    super();
    /** @type {number} */
    this._callCount = 0;
  }

  /**
   * @param {number} amount
   * @param {string} currency
   * @returns {PaymentResult}
   */
  charge(amount, currency) {
    this._callCount++;
    return new PaymentResult(true, `mock_tx_${this._callCount}`, `Mock charge ${amount} ${currency}`);
  }

  /**
   * @param {string} transactionId
   * @returns {PaymentResult}
   */
  refund(transactionId) {
    return new PaymentResult(true, `mock_ref_${transactionId}`, `Mock refund ${transactionId}`);
  }

  /** @returns {number} */
  getCallCount() { return this._callCount; }
}

File: skill_sdk/tests/fixtures/cpp-project/src/shop/MockGateway.cpp
#include "MockGateway.h"

MockGateway::MockGateway() {}

PaymentResult MockGateway::Charge(double amount, const std::string& currency) {
    callCount_++;
    return PaymentResult{
        true,
        "mock_tx_" + std::to_string(callCount_),
        "Mock charge " + std::to_string(amount) + " " + currency
    };
}

PaymentResult MockGateway::Refund(const std::string& transactionID) {
    return PaymentResult{
        true,
        "mock_ref_" + transactionID,
        "Mock refund " + transactionID
    };
}
